optimism/src/lib.rs

```diff
@@ -29,4 +29,32 @@ pub mod proof;
 /// The RAM lookup argument.
 pub mod ramlookup;
 
+use kimchi::circuits::expr::{ConstantExpr, Expr};
+use kimchi_msm::{columns::Column, witness::Witness};
+use lookups::Lookup;
+use std::collections::HashMap;
+
 pub use ramlookup::{LookupMode as RAMLookupMode, RAMLookup};
+
+/// Type to represent a constraint on the individual columns of the execution
+/// trace.
+/// As a reminder, a constraint can be formally defined as a multi-variate
+/// polynomial over a finite field. The variables of the polynomial are defined
+/// as `kimchi_msm::columns::Column``.
+/// The `expression` framework defined in `kimchi::circuits::expr` is used to
+/// describe the multi-variate polynomials.
+/// For instance, a vanilla 3-wires PlonK constraint can be defined using the
+/// multi-variate polynomial of degree 2
+/// `P(X, Y, Z) = q_x X + q_y Y + q_m X Y + q_o Z + q_c`
+/// To represent this multi-variate polynomial using the expression framework,
+/// we would use 3 different columns.
+pub(crate) type E<F> = Expr<ConstantExpr<F>, Column>;
+
+pub(crate) struct Circuit<const N: usize, SELECTOR, F> {
```

Comment:
This is the definition of a Circuit, and can be instantiated for MIPS with SELECTOR=Instruction and Keccak with SELECTOR=Steps

Reply:
Will probably talk to Danny about this to figure out how easy it is to generalize / make uniform across projects. For now, I just used it as a local struct for testing purposes.